photon-ml/src/test/scala/com/linkedin/photon/ml/optimization/LinearRegressionOptimizationProblemTest.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.optimization
+
+import com.linkedin.photon.ml.data.{LabeledPoint, SimpleObjectProvider}
+import com.linkedin.photon.ml.model.Coefficients
+import com.linkedin.photon.ml.normalization.NormalizationContext
+import com.linkedin.photon.ml.optimization.game.GLMOptimizationConfiguration
+import com.linkedin.photon.ml.test.CommonTestUtils
+
+import org.apache.spark.rdd.RDD
+import org.mockito.Mockito._
+import org.testng.Assert._
+import org.testng.annotations.Test
+
+class LinearRegressionOptimizationProblemTest {
+  import LinearRegressionOptimizationProblemTest._
+  import CommonTestUtils._
+
+  @Test
+  def testUpdateObjective(): Unit = {
+    val problem = createProblem
+    val normalizationContext = new SimpleObjectProvider(mock(classOf[NormalizationContext]))
+    val regularizationWeight = 1D
+
+    assertNotEquals(problem.regularizationWeight, regularizationWeight)
+
+    val updatedProblem = problem.updateObjective(normalizationContext, regularizationWeight)
+
+    assertEquals(updatedProblem.regularizationWeight, regularizationWeight)
+  }
+
+  @Test
+  def testInitializeZeroModel(): Unit = {
+    val problem = createProblem
+
+    val dim = 10
```

Comment:
May as well pull dim out and into the object.

In addition, maybe initializeZeroModel and createModel tests should check the class type of the model?